cdap-api-spark/src/main/java/co/cask/cdap/api/spark/JavaSparkExecutionContext.java

```diff
@@ -133,59 +142,73 @@
                                                        @Nullable Iterable<? extends Split> splits);
 
   /**
-   * Creates a {@link JavaPairRDD} that represents all events from the given stream as (timestamp, string body) pair.
+   * Creates a {@link JavaRDD} that represents all events from the given stream.
    *
    * @param streamName name of the stream
-   * @return A new {@link JavaPairRDD} instance with the key as the stream event timestamp and value is the
-   *         UTF-8 decoded event body
+   * @return A new {@link JavaRDD} instance that reads from the given stream
    * @throws DatasetInstantiationException if the Stream doesn't exist
    */
-  public JavaPairRDD<Long, String> fromStreamAsStringPair(String streamName) {
-    return fromStreamAsStringPair(streamName, 0, Long.MAX_VALUE);
+  public JavaRDD<StreamEvent> fromStream(String streamName) {
+    return fromStream(streamName, 0, Long.MAX_VALUE);
   }
 
   /**
-   * Creates a {@link JavaPairRDD} that represents events from the given stream as (timestamp, string body) pair
-   * in the given time range.
+   * Creates a {@link JavaRDD} that represents events from the given stream in the given time range.
    *
    * @param streamName name of the stream
    * @param startTime the starting time of the stream to be read in milliseconds (inclusive)
    * @param endTime the ending time of the streams to be read in milliseconds (exclusive)
-   * @return A new {@link JavaPairRDD} instance that reads from the given stream,
-   *         with the stream event timestamp as key and the UTF-8 decoded event body as value
+   * @return A new {@link JavaRDD} instance that reads from the given stream
```

Comment:
is there a way to get all event up to, or all events since a given time? I guess you pass in 0 for the start or Long.MAX_VALUE for the end, perhaps that should be mentioned in the Javadoc.